chronos/asyncsync.nim

```diff
@@ -70,24 +70,22 @@ proc newAsyncLock*(): AsyncLock =
   # Workaround for callSoon() not worked correctly before
   # getGlobalDispatcher() call.
   discard getGlobalDispatcher()
-  result = new AsyncLock
-  result.waiters = newSeq[Future[void]]()
-  result.locked = false
+  AsyncLock(waiters: newSeq[Future[void]](), locked: false)
 
-proc wakeUpFirst(lock: AsyncLock) {.inline.} =
+proc wakeUpFirst(lock: AsyncLock): bool {.inline.} =
   ## Wake up the first waiter if it isn't done.
   for fut in lock.waiters.mitems():
     if not(fut.finished()):
       fut.complete()
-      break
+      return true
```

Comment:
best would be to remove the finished futures from the waiters list here since they no longer serve any purpose in the list